app/controllers/posts_controller.rb

```diff
@@ -29,6 +32,12 @@ def show
     @post = Post.find(params[:id])
     @posts = @user.posts.find(params[:id])
     @comments = Comment.where(post_id: params[:id])
+    respond_to do |format|
+      format.html
+      format.xml { render xml: @post }
+      format.json { render json: @post }
+      format.json { render json: @comments }
+    end
```

Comment:
- You did a great job by converting the response into JSON format but make sure that you have  API endpoint that returns all posts in JSON format- **GET:** `/api/users/:user_id/posts` or `/api/v1/users/:user_id/posts` as required in the project . try to test your points before submission so you will be sure there are working as expected. you can use thunder client or postman to test them :+1: